tools/linkage_disequilibrium/ld_tool

```diff
@@ -0,0 +1,401 @@
+    }
+  }
+
+  if (@regions) {
+    if (!$config->{calculation} eq 'region') {
+      die "Region input must be used with --calculation region."; 
+    }
+  }
+
+  if (@variants) {
+    if (!($config->{calculation} eq 'center' || $config->{calculation} eq 'pairwise')) {
+      die "Variant input must be used with --calculation center or --calculation pairwise."; 
+    }
+  }
+
+}
+
+sub init_connections {
+  my $config = shift;
+  my $registry = 'Bio::EnsEMBL::Registry';
+  my $port = 3306;
+  my $assembly = lc $config->{assembly};
+  if ($assembly eq 'grch37') {
+    $port = 3337;
+  }
+  $registry->load_registry_from_db(
+    -host => 'ensembldb.ensembl.org',
+    -user => 'anonymous',
+    -port => $port,
+  );
+
+  my $species = $config->{species};
+  my $population_adaptor = $registry->get_adaptor($species, 'variation', 'population');
+  my $variation_adaptor = $registry->get_adaptor($species, 'variation', 'variation');
+  my $ld_feature_container_adaptor = $registry->get_adaptor($species, 'variation', 'ldfeaturecontainer');
+  $ld_feature_container_adaptor->db->use_vcf(1);
+  my $d_prime = $config->{'d_prime'};
+  if ($d_prime && ($d_prime < 0.0 || $d_prime > 1.0)) {
+    die "d_prime must be in the range of 0.0 <= d_prime <= 1.0";
+  }
+  my $r2 = $config->{'r2'};
+  if ($r2 && ($r2 < 0.0 || $r2 > 1.0)) {
+    die "r2 must be in the range of 0.0 <= r2 <= 1.0";
+  }
+  $ld_feature_container_adaptor->min_d_prime($d_prime) if (defined $d_prime);
+  $ld_feature_container_adaptor->min_r2($r2) if (defined $r2);
+
+  my $slice_adaptor = $registry->get_adaptor($species, 'core', 'slice');
+ 
+  $config->{population_adaptor} = $population_adaptor;
+  $config->{variation_adaptor} = $variation_adaptor;
+  $config->{ld_feature_container_adaptor} = $ld_feature_container_adaptor; 
+  $config->{slice_adaptor} = $slice_adaptor;
+}
+
+sub show_populations {
+  my $config = shift;
+  my $population_adaptor = $config->{population_adaptor}; 
+  if ($config->{species} eq $DEFAULT_SPECIES) {
+    my @super_populations = qw/1000GENOMES:phase_3:AFR 1000GENOMES:phase_3:AMR 1000GENOMES:phase_3:EAS 1000GENOMES:phase_3:EUR 1000GENOMES:phase_3:SAS/;
+    print "Choose any of the populations for LD computation. If you choose a super population all sub populations will be included.\n";
+    foreach my $name (@super_populations) {
+      my $population = $population_adaptor->fetch_by_name($name);
+      print $population->name, ' ', $population->description, " (super population)\n"; 
+      foreach my $sub_population (sort {$a->name cmp $b->name} @{$population->get_all_sub_Populations}) {
+        print ' ', $sub_population->name, ' ', $sub_population->description, "\n";
+      }
+    }   
+  } else {
+    my $populations = $population_adaptor->fetch_all_LD_Populations;  
+    foreach my $population (sort {$a->name cmp $b->name} @$populations) {
+      print $population->name, ' ', $population->description, "\n";
+    }
+  }
+}
+
+sub init_populations {
```

Comment:
We should check at least one valid population is supplied and give a suitable warning if not.